### PR TITLE
feat(ucmc-web): gear loans + checkout

### PR DESCRIPTION
**Pull request by SpencerAWill**
## Summary

Adds the gear loan / checkout flow on top of the gear-inventory feature (#86), plus a **bulk backfill** for migrating historical loans from paper logbooks. Officer-driven: a gear-cave keeper checks pieces out to a member from the **gear-desk Sheet** (one Sheet, mode toggle at the top), and check-in batches may span multiple borrowers (each row resolves its own loan from the gear code). Members see their own active + history loans at `/my/gear`.

### Schema + RBAC
- New `gear_loans` table (one row = one piece on loan to one member) with a partial unique on `(gear_id) WHERE returned_at IS NULL` as the race protector against two officers checking out the same piece simultaneously. FK choices: `gear` / `memberUserId` are `RESTRICT` (can't retire on-loan gear or delete a member with open loans); `checkedOutBy` / `returnedTo` are `SET NULL` so closed-loan history survives officer churn.
- Three new `auditAction` values: `loan.checked_out` (one per row, `bulk: true`), `loan.checked_in`, `loan.extended`.
- New `gear:loan` permission (officer-only, distinct from `gear:manage`) so the desk authority can be delegated independently of inventory CRUD. `system_admin` gets it free via the role bypass.

### Server actions
- `checkoutLoansAction` — one-member-many-items, race-safe: tries the bulk insert first, falls back to per-row on partial-index violation so winners still land and the loser is reported as `already_on_loan`.
- `checkinLoansAction` — any-items-any-members; each row resolves its own loan.
- `extendLoanAction`, `listLoansAction`, `getLoanDetailAction`, `listMyLoansAction`, plus three lookup helpers for the gear-desk pickers.
- `retireGearAction` (single + bulk) consults the open-loan table and returns typed `on_loan` instead of NULLing a code mid-loan. Bulk variant batches the UPDATE + audit INSERT into one `db.batch`.
- **`bulkImportLoansAction` (backfill)** — accepts ≤500 rows keyed on member primary email + gear `code`. Supports both open and pre-returned rows in the same import; eligibility checks for `lifecycle` and `condition` are intentionally relaxed (a piece retired today may have been serviceable when loaned years ago). Audit events carry `bulk: true, backfill: true`. Does NOT mutate `gear.condition` from `condition_at_return`.

### UI
- **`/gear/loans`** — officer list with active/history tabs, member filter combobox (URL-driven, rehydrates the chip on refresh), overdue-only toggle, sort, full shadcn paginator (always rendered while ≥1 row). Header has the "Gear desk" button, a "Backfill" button, and a mobile-tight always-visible icon button.
- **`/gear/loans/$publicId`** — loan detail card with borrower `UserAvatar` and an `Extend` dialog.
- **`/my/gear`** — member's own active loans on top, history below. Reachable from the user-menu "My Gear" item.
- **Gear-desk Sheet** — single Sheet with checkout / check-in tab toggle. Each pane adds items by inline barcode scanner (sticky viewfinder, sticky on both breakpoints) or by code-search `Command` combobox at the bottom of the items list. Item rows are a shadcn `<Table>` with column alignment; check-in rows show a hover-tooltip `UserAvatar` for the borrower. Check-in notes live in a `StickyNote`-icon Dialog per row so they don't bloat the table.
- **Loans Bulk Import Sheet** — file picker + clipboard paste + manual row entry, mirroring `GearBulkImportSheet`. Per-row preview shows skip reasons (`member_not_found`, `gear_not_found`, `already_on_loan`, `invalid_dates`, `duplicate_in_import`).

### Barcode scanner
- Hand-rolled hybrid: browser-native `BarcodeDetector` API where available (Chrome / Edge / Android Chrome / Safari iOS 17+ / Safari macOS 17+ — zero deps, zero WASM); falls back to the `barcode-detector` ponyfill with **locally-vendored** `zxing-wasm` at `public/zxing-wasm/zxing_reader.wasm` so `connect-src 'self'` is enough.
- Camera picker with localStorage-persisted `deviceId` for gear-cave laptops with USB cameras.
- `OverconstrainedError` (unplugged USB cam) auto-retries with default constraints before surfacing an error.
- CSP: `script-src 'wasm-unsafe-eval'` and `Permissions-Policy: camera=(self)` added in `src/server/headers.server.ts`.

### Privacy
- `getGearDetailAction` joins the currently-open loan and strips `memberFullName` for callers who lack `gear:loan` AND aren't the borrower themselves.
- `/my/gear` filters server-side on `memberUserId === principal.userId`.

### Dependency hygiene
- `protobufjs` overridden to `>=8.0.2` to clear 7 transitive Dependabot advisories (4 high, 3 moderate) from `@pulumi/pulumi`'s grpc-js chain. Touches infra-only deps; web tree unaffected.

### Tests
- **94 server-pool tests** covering RBAC, happy paths, per-row skip reasons, multi-borrower check-in, condition update + audit fan-out, extension edge cases, retire-block-on-loan, race-replay (mocks `getOpenLoanForGear` so the partial unique index trips on the bulk insert), member lookup by primary email, and 13 backfill tests across happy paths + every skip reason.
- 46 dom-pool tests pass.
- New e2e `gear-loans.spec.ts` exercises the officer-checkout-then-checkin flow ending in the member's `/my/gear` history (camera path skipped — headless Chromium camera permissions are unreliable; the code-search combobox covers the same resolve pipeline).

### Docs
- README + CLAUDE.md gear-loans paragraphs updated (the latter under "Web app load-bearing invariants"), including the backfill bullet.

## Test plan

- [x] `pnpm --filter ucmc-web typecheck` (clean)
- [x] `pnpm exec eslint apps/web` (no new errors)
- [x] `pnpm --filter ucmc-web test` (94 server + 46 dom = 140 / 140)
- [x] `pnpm --filter ucmc-web db:migrate:local` (0034 + 0035 applied)
- [x] `pnpm audit --audit-level=moderate --prod` clean after protobufjs override
- [ ] Manual: officer → `/gear/loans` → Gear desk → search member → scan/search two codes → Check out → toast + rows on `/gear/loans` and member's `/my/gear`
- [ ] Manual: retire one of the on-loan pieces from `/gear` → blocked with `on_loan` toast
- [ ] Manual: Gear desk Check-in tab → set `Condition at return` → submit → loan closes, gear condition updates, audit reflects both events
- [ ] Manual: backfill sheet — upload a CSV with mixed open/returned rows, paste from clipboard, manual row entry; confirm skip reasons render
- [ ] Manual: USB camera unplug → reopen scanner → falls back to built-in camera without flashing error
- [ ] e2e `gear-loans.spec.ts`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,7 +48,8 @@
       "esbuild@<0.25.0": ">=0.25.0",
       "fast-uri@<3.1.2": ">=3.1.2",
       "ip-address@<10.1.1": ">=10.1.1",
-      "protobufjs@<7.5.5": ">=7.5.5"
+      "protobufjs@<7.5.5": ">=7.5.5",
+      "protobufjs@<8.0.2": ">=8.0.2"
```
[the remark this answers is not in the document]
Comment (SpencerAWill) [reply package.json:51]: Done in a689461 — dropped the redundant `<7.5.5` entry; the `<8.0.2` rule subsumes it. Lockfile still resolves protobufjs to 8.2.0.

File: apps/web/src/features/gear/server/loans-repo.server.ts [hunk 2]
```diff
@@ -0,0 +1,579 @@
...
+    clauses.push(sql`${schema.gearLoans.dueAt} < (unixepoch() * 1000)`);
+  }
+  if (options.q && options.q.trim().length > 0) {
+    const needle = `%${escapeLikeNeedle(options.q.trim())}%`;
+    clauses.push(
+      or(
+        like(schema.gear.code, needle),
+        like(schema.gear.description, needle),
+        like(schema.profiles.fullName, needle),
+      ),
```
[the remark this answers is not in the document]
Comment (SpencerAWill) [reply apps/web/src/features/gear/server/loans-repo.server.ts:218]: Implemented in a689461 — joined `user_emails` (primary only) on both the SELECT and COUNT queries and added `like(userEmails.email, needle)` to the OR clause. New test asserts full-email and local-part substring both match.

File: apps/web/src/server/headers.server.ts [hunk 3]
```diff
@@ -53,8 +61,12 @@ export const SECURITY_HEADERS: ReadonlyArray<readonly [string, string]> = [
...
   ["Referrer-Policy", "strict-origin-when-cross-origin"],
   [
+    // Camera is allowed on the same origin only — the gear-cave
+    // checkout flow opens the rear camera via getUserMedia +
+    // BarcodeDetector to scan gear labels. Microphone / geolocation /
+    // payment stay disabled (no feature uses them).
     "Permissions-Policy",
-    "camera=(), microphone=(), geolocation=(), payment=()",
+    "camera=(self), microphone=(), geolocation=(), payment=()",
   ],
```
[the remark this answers is not in the document]
Comment (SpencerAWill) [reply apps/web/src/server/headers.server.ts]: Scoped in a689461. `Permissions-Policy` is now computed per-request via `securityHeadersForPath(url.pathname)` called from the start.ts middleware; only `/gear/loans*` gets `camera=(self)`, everything else gets `camera=()`. Unit test in `apps/web/src/server/__tests__/headers.test.ts` pins the matrix.

File: apps/web/e2e/gear-loans.spec.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+  await waitForHydration(page);
+  await page.getByRole("button", { name: /continue to ucmc/i }).click();
+  await page.waitForURL((u) => !u.pathname.startsWith("/auth/callback"), {
+    timeout: 15_000,
+  });
+
+  // Open /gear/loans and the gear-desk Sheet.
+  await page.goto("/gear/loans");
+  await waitForHydration(page);
+  await page.getByRole("button", { name: /^gear desk$/i }).click();
```
[the remark this answers is not in the document]
Comment (SpencerAWill) [reply apps/web/e2e/gear-loans.spec.ts]: Fixed in a689461 — selector now matches the aria-label `Open gear desk` instead of the visible text (which collapses on narrow widths via `sm:inline`).

File: apps/web/e2e/gear-loans.spec.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+  await page
+    .getByRole("option", { name: /e2e tester/i })
+    .first()
+    .click();
+
+  for (const code of [code1, code2]) {
+    await page.getByRole("button", { name: /search code/i }).click();
+    await page.getByPlaceholder(/enter code/i).fill(code);
+    await page.getByRole("option", { name: new RegExp(code) }).click();
+  }
```
[the remark this answers is not in the document]
Comment (SpencerAWill) [reply apps/web/e2e/gear-loans.spec.ts:95]: Fixed in a689461 — the `Search code` button no longer exists (the always-visible `GearCodeSearchCombobox` replaced it). The spec now fills the inline `CommandInput` by placeholder (`Enter code (CH1, LJ3…) and press Enter`) and selects the resulting option directly.

File: apps/web/e2e/gear-loans.spec.ts [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+  await page.getByRole("button", { name: /^check out 2 items$/i }).click();
+  await expect(
+    page.getByText(new RegExp(`checked out 2 pieces`, "i")),
+  ).toBeVisible({ timeout: 5_000 });
+
+  // CHECK IN: switch modes inside the same Sheet, scan both back in.
+  await page.getByRole("button", { name: /^gear desk$/i }).click();
+  await page.getByRole("tab", { name: /check in/i }).click();
+  for (const code of [code1, code2]) {
+    await page.getByRole("button", { name: /search code/i }).click();
```
[the remark this answers is not in the document]
Comment (SpencerAWill) [reply apps/web/e2e/gear-loans.spec.ts]: Fixed in a689461 — same shape as the checkout fix. The check-in pane has an always-visible `GearCodeSearchCombobox` (placeholder `Enter code to check in…`), so the spec fills it directly and clicks the resulting option.

File: CLAUDE.md [hunk 7]
```diff
@@ -90,6 +90,7 @@ Email three-tier fallback (`src/server/email/resend.ts`): (1) Resend API if `RES
...
...ton and a mobile FAB on `/gear/loans`. Barcode scanning uses **`@yudiel/react-qr-scanner`** (CODE128 + QR whitelist, native `BarcodeDetector` on iOS 17+ / Android Chrome, polyfilled elsewhere) — lazy-loaded so the library only ships on `/gear/loans`. New audit actions: `loan.checked_out` (one event per row in a batch, with `bulk: true`), `loan.checked_in`, `loan.extended`. **Backfill** lives in the `LoansBulkImportSheet` (header "Backfill" button on `/gear/loans`, gated by `gear:loan`) — `bulkImportLoansAction` reads CSV (file / clipboard / manual rows), resolves members by primary email through `user_emails` and gear by `code`, supports both open and pre-returned rows in the same import. Eligibility checks for `lifecycle` and `condition` are intentionally relaxed (a piece retired today may have been serviceable when loaned years ago); the partial unique index still gates open backfill rows, surfaced as `already_on_loan` skip. Audit events for backfilled rows carry `bulk: true, backfill: true`; closed-loan rows also emit a `loan.checked_in` in the same call. Backfill does NOT mutate `gear.condition` from `condition_at_return` — the historical condition belongs on the loan row only.
```
[the remark this answers is not in the document]
Comment (SpencerAWill) [reply CLAUDE.md]: Fixed in 7b8388a. The CLAUDE.md gear-loans paragraph now describes the real implementation (native `BarcodeDetector` + `barcode-detector/ponyfill` fallback, locally-vendored ZXing WASM, single responsive header button — no FAB).

File: apps/web/src/features/gear/server/loans-bulk-import-actions.server.ts [hunk 8]
```diff
@@ -0,0 +1,311 @@
...
+ *      loan in the same insert — no follow-up check-in is needed.
+ *      Open backfill rows are still subject to the
+ *      `gear_loans_one_active_per_gear` partial unique index, which
+ *      surfaces as a per-row `already_on_loan` skip.
+ *
+ * Inserts run sequentially because audit emission is one event per
+ * row and we want the same per-row skip surfacing as
+ * `bulkImportGearAction` — a single batch failure on row 7 would
+ * roll back the first 6 too. ≤200 rows is fine at one round-trip
+ * each for a one-time backfill flow.
```
[the remark this answers is not in the document]
Comment (SpencerAWill) [reply apps/web/src/features/gear/server/loans-bulk-import-actions.server.ts]: Aligned in 7b8388a. The module docstring now references the real 500-row cap (zod `bulkImportLoansInputSchema.max(500)` mirrored by the sheet `MAX_ROWS`).

File: apps/web/src/features/gear/components/my-gear-list.tsx [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+ * rather than something the member acts on.
+ */
+export function MyGearList() {
+  const { data, isLoading } = useQuery(myLoansQueryOptions());
+  if (isLoading) {
+    return <p className="text-sm text-muted-foreground">Loading your gear…</p>;
+  }
+  const active = data?.active ?? [];
+  const history = data?.history ?? [];
+
```
[the remark this answers is not in the document]
Comment (SpencerAWill) [reply apps/web/src/features/gear/components/my-gear-list.tsx:41]: Fixed in 7b8388a. `MyGearList` now destructures `isError` / `error` from `useQuery` and renders a destructive `<Alert>` with the error message (or a generic fallback) instead of falling through to the empty state.

File: apps/web/src/features/gear/components/gear-desk-sheet.tsx [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+import { GearDeskCheckinPane } from "#/features/gear/components/gear-desk-checkin-pane";
+import { GearDeskCheckoutPane } from "#/features/gear/components/gear-desk-checkout-pane";
+
+type Mode = "checkout" | "checkin";
+
+/**
+ * Gear-cave action surface — one Sheet that hosts both the checkout
+ * and check-in flows, with a tab toggle at the top. Opens from a
+ * "Gear desk" button in the page header (and a FAB on mobile —
+ * `GearDeskTrigger` handles both placements).
```
[the remark this answers is not in the document]
Comment (SpencerAWill) [reply apps/web/src/features/gear/components/gear-desk-sheet.tsx]: Fixed in 7b8388a. The `GearDeskSheet` docstring now describes the single always-visible responsive header button and notes the FAB iteration was dropped (the FAB fought with the sidebar stacking context).
